File: algo-fibonacci/fibonacci.js
function fibonacci(num) {
  // Coverage for first two nums of sequence
  if (num === 0) {
    return 0;
  } else if (num === 1) {
    return 1;
  }

  // Set first nums of fibonacci sequence
  let twoBack = 0;
  let oneBack = 1;

  // Placeholder
  let current = 0;

  // Iterate num amount of times for nth term in sequence past 1
  for (let i = 1; i < num; i++) {
    current = oneBack + twoBack;
    twoBack = oneBack;
    oneBack = current;
  }

  return current;
}

module.exports = fibonacci;

File: algo-fibonacci-py/fibonacci.py
def fibonacci(num):
  # Edge cases
  if num == 0:
    return 0
  elif num == 1:
    return 1

  # Set first terms
  twoBack = 0
  oneBack = 1
  current = 0

  # Iterate over n sequences past 2nd term
  for x in range(num, 1, -1):
    current = oneBack + twoBack
    twoBack = oneBack
    oneBack = current

  return current
